Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift

```diff
@@ -156,12 +195,52 @@ open class ChatMessageListView: UITableView, Customizable, ComponentsProvider {
         with changes: [ListChange<ChatMessage>],
         completion: (() -> Void)? = nil
     ) {
-        listChangeUpdater.performUpdate(with: changes) { _ in
-            completion?()
+        let bottomChange = changes.first(where: { $0.indexPath.item == 0 })
+        if !isLastCellFullyVisible && bottomChange?.isInsertion == true && bottomChange?.item.isSentByCurrentUser == false {
+            changes.filter(\.isInsertion).forEach {
+                skippedMessages.insert($0.item.id)
+            }
+            // Don't update UI if we are skipping the inserted messages
```

Comment:
If we handle it like this, then when a new message arrives, the reactions of the visible messages are not reflected in the UI. We need to apply the updates in cases like this, just ignore the inserts.

Reply:
Initially, I had it like you are saying, but it was causing some problems from what I can remember. So I decided to make it more simple. The reality is that usually, this is not a visible problem, because the messages being added and updated are 90% of the time out of the screen when the index == 0. But I'll double-check this one and see what I can do 👍

Reply:
The main issue here is that when I've scrolled up and read older messages, if I react to one of them, there's no visual feedback. And the use-case is pretty common, so I think it's worth revisiting this.

Reply:
When you scroll down to read older messages, the cell will be reused and the data will be updated by that time 🤔 But I'll investigate this one a little bit more too and will comeback with feedback soon 👍

Reply:
I was able to understand the problem and now it is fixed, all updates are still executed, but the insertions :)